.changeset/big-buttons-shop.md

```diff
@@ -0,0 +1,7 @@
+---
+'@astrojs/starlight': minor
+---
+
+Removes the `/` search shortcut for [accessibility reasons](https://www.w3.org/WAI/WCAG21/Understanding/character-key-shortcuts.html).
+
+⚠️ **Potentially breaking change:** The `search.shortcutLabel` UI string has been removed. If you were using this string in your custom UI, you will need to update your code.
```

Comment:
That's a good question. I'll do some research on this as this is not something I have an answer for off the top of my head (maybe could even be something to ask to the i18n teams).

Reply:
Small update based on a bit of research and some [feedback on Discord](https://discord.com/channels/830184174198718474/867764597838184478/1258435440671719506): I think we probably should provide a mechanism to set localized versions. In theory there are three labels needed I think?

- “Command” — accessible label
- “Control” — accessible label
- “Ctrl” — visible key label

I guess allowing overriding `⌘` could also be needed in theory, but in practice I haven’t seen much evidence for that being displayed different.

As an example based on [German Wikipedia information](https://de.wikipedia.org/wiki/Tastenkombination#Hilfstasten_f%C3%BCr_Tastenkombinationen) and a [comment on Discord](https://discord.com/channels/830184174198718474/867764597838184478/1258437022616649811), in German this might look like:

```json
{
  "search.ctrlKey": "Strg",
  "search.ctrlLabel": "Steuerungstaste",
  "search.cmdLabel": "Befehlstaste"
}
```

Reply:
Oh and in case it isn’t clear, that can always happen in a follow-up PR if we want. It seems like a relatively small portion of people who would be impacted.

Reply:
I just updated the PR and only ended up adding `search.ctrlKey`.

The reasoning behind this decision is that depending on the screen-reader, 2 differents strings can be read, depending on the `aria-keyshortcuts` support of the screen-reader.

- When the screen-reader does not support `aria-keyshortcuts`, the string read will use `search.ctrlKey` used in the `<kbd>` element.
- When the screen-reader supports `aria-keyshortcuts`, according to the [spec](https://www.w3.org/TR/wai-aria-1.3/#aria-keyshortcuts), _"Authors MUST specify modifier keys exactly according to the UI Events KeyboardEvent key Values [spec](https://www.w3.org/TR/uievents-key/#keys-modifier) - for example, "Alt", "Control", "Shift", "Meta", or "AltGraph"."_ From my understanding, this means that if any internationalization is needed, it should be done by the screen-reader itself, and not by the author which cannot provide any translation for a modifier key.

Note that this _may_ change in the future as the internationalization (I18N) working group [reviewed](https://www.w3.org/2024/03/21-aria-minutes#t06) fairly recently the `aria-keyshortcuts` attribute and may raise some related issues in future meetings.

Let me know if you have any concerns about this decision.

Reply:
Sounds good to me! Thanks for figuring that out 💖